MDANSE_GUI/Src/MDANSE_GUI/MolecularViewer/MolecularViewer.py

```diff
@@ -274,270 +252,485 @@ def add_text(text, coord):
+
+    def _new_trajectory_object(self, fname: str, trajectory: Trajectory):
+        """Creates and sets a new trajectory reader for the input trajectory.
+
+        Parameters
+        ----------
+        fname : str
+            trajectory file name
+        data : Trajectory
+            instance of the MDANSE input trajectory handler
+        """
+        reader = hdf5wrapper.HDF5Wrapper(fname, trajectory, trajectory.chemical_system)
+        if (self._reader is not None) and (reader.filename == self._reader.filename):
+            return
+        self.set_reader(reader)
+
+    @Slot(float)
+    def _new_scaling(self, scale_factor: float):
+        """Update the scale factor by which all the atom radii are multiplied.
+
+        Scale factor 1.0 means that the covalent radii of atoms are used as
+        radii of the spheres in the 3D view. By default the atom size is scaled
+        down to allow the user to see atoms behind the first layer and the
+        bonds between atoms.
+
+        Parameters
+        ----------
+        scale_factor : float
+            Sphere radii in 3D view will be multiplied by this factor
+        """
+        self._scale_factor = scale_factor
+        if self._reader is not None:
+            self.create_atoms()
+            self.update_renderer()
+
+    def _new_visibility(self, flags: list[bool]):
+        """Takes the new values of boolean flags which make
+        different actors in the 3D scene (in)visible.
+
+        Parameters
+        ----------
+        flags : List[bool]
+            Each actor will be visible if its flag is True.
+        """
+        self._atoms_visible = flags[0]
+        self._bonds_visible = flags[1]
+        self._cell_visible = flags[2]
+        self.create_unit_cell()
+        self.update_atm_polydata()
+        self.create_atoms()
+        self.create_bonds()
+        self.update_renderer()
+
+    def trace_from_dialog(self, params: dict[str, Any]):
+        """Passes the input parameter dictionary to the method
+        which draws an isosurface in the 3D view.
+
+        Parameters
+        ----------
+        params : Dict[str, Any]
+            dictionary of input parameters from TraceWidget.py
+        """
+        self._draw_isosurface(params["atom_number"], params)
+
+    def delete_isosurface_from_dialog(self, trace_number: int):
+        """Deletes from the 3D scene the isosurface with a specified
+        index, if it exists.
+
+        Parameters
+        ----------
+        trace_number : int
+            index of the isosurface
+        """
+        try:
+            surface = self._surfaces[trace_number]
+        except IndexError:
+            return
+        else:
+            surface.VisibilityOff()
+            surface.ReleaseGraphicsResources(self._iren.GetRenderWindow())
+            self._renderer.RemoveActor(surface)
+            self._surfaces.pop(trace_number)
+            self.update_renderer()
+            self.changed_trace.emit()
```

Comment:
Don't need `else` after `return`
```suggestion
        surface.VisibilityOff()
        surface.ReleaseGraphicsResources(self._iren.GetRenderWindow())
        self._renderer.RemoveActor(surface)
        self._surfaces.pop(trace_number)
        self.update_renderer()
        self.changed_trace.emit()
```